### PR TITLE
chore: upgrade to Go 1.20

### PR DESCRIPTION
**Pull request by rootulp**
Closes https://github.com/celestiaorg/celestia-app/issues/1691

### CODE REVIEW

File: pkg/shares/parse_test.go [hunk 1]
```diff
@@ -220,13 +220,14 @@ func generateRawShare(namespace appns.Namespace, isSequenceStart bool, sequenceL
...
+	return padWithRandomBytes(t, rawShare)
 }
 
-func padWithRandomBytes(partialShare []byte) (paddedShare []byte) {
+func padWithRandomBytes(t *testing.T, partialShare []byte) (paddedShare []byte) {
 	paddedShare = make([]byte, appconsts.ShareSize)
 	copy(paddedShare, partialShare)
-	rand.Read(paddedShare[len(partialShare):])
+	_, err := rand.Read(paddedShare[len(partialShare):])
+	assert.NoError(t, err)
```
Comment (cmwaters) [inline pkg/shares/parse_test.go:230]: nit: I'd generally `require` no error here. This is bootstrapping code. There is no point in continuing if this errors for some reason

**Review by cmwaters — APPROVED** [review]
LGTM

File: pkg/shares/parse_test.go (hunk 1 above)
Comment (rootulp) [reply pkg/shares/parse_test.go]: resolved by https://github.com/celestiaorg/celestia-app/pull/1692/commits/7155cb503ff9677571c759fee0ecaa92f3dd2974

**Review by evan-forbes — APPROVED** [review]
sorry, didn't approve yesterday cause a bunch of CI was failing due to the github outage. reran CI and its all fine

**Review by cmwaters — APPROVED** [review]

**Pull request merged**